### PR TITLE
feat(common): introduce ngAttr directive

### LINKED ISSUE #13940: Add `ngAttr` like `ngClass` and `ngStyle`
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report
[x] feature request
[ ] support request 
```

possibly related to #13355. might help with #9121

**Current behavior**
There is an `ngStyle` that takes an object whose key/values add a style for each key and set to the corresponding value.

There is nothing comparable for attributes.  There is `[attr.xxx]` which adds-and-sets the `xxx` attribute. Unfortunately, if you have a lot of attributes to set, you'll add a lot of `[attr]` bindings. And you could have a lot of the to set when enriching the app with ARIA.

**Expected behavior**

A comparable `ngAttr` directive would do the trick.

**Minimal reproduction of the problem with instructions**

This plunker provides a suggested implementation of `NgAttr` that borrows heavily from `NgStyle`

https://plnkr.co/edit/gcoALdBgNwqgrRtWfvr2?p=preview

**What is the motivation / use case for changing the behavior?**

Make it easier to add (and remove) a large number of ARIA attributes.

Angular feels incomplete without it as there is `[class.xxx]`/`ngClass`, `[style.xxx]`/`NgStyle`, but only `[attr.xxx]`.

**Please tell us about your environment:**
All

* **Angular version:** 2.0.X
2.0.4

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web 
all

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes #13940

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@DzmitryShylovich you've forgotten to commit unit tests :-) 

Before doing more work on this [I would like to hear a bit more](https://github.com/angular/angular/issues/13940#issuecomment-273228917) about real-life scenarios, though.

**Comment by DzmitryShylovich** [comment]
> Before doing more work on this I would like to hear a bit more about real-life scenarios, though.

I have doubts too, this is why I didn't add tests :)

**Comment by mhevery** [comment]
What happens if you have:
```
<div [foo]="exp1" [ngAttr="{ foo: exp2 }">
```

There is no easy way to detect that. 

I would also like to see real world scenarios. Please update or close this PR

**Comment by DzmitryShylovich** [comment]
@mhevery Take a look at https://github.com/angular/angular/issues/13940#issuecomment-273343339

**Comment by pkozlowski-opensource** [comment]
@DzmitryShylovich @mhevery I've updated the original issue with my thoughts on the `ngAttr`: https://github.com/angular/angular/issues/13940#issuecomment-276079860

tl;dr; is that IMO it is not adding enough benefit to "pay" for itself. Having said this I would prefer that @mhevery decides here.

**Comment by mhevery** [comment]
My experience tells me that more things means more bugs and supporting voes. So I vote against this as well. Going to close the PR for now.

**Comment by nayfin** [comment]
I am working on a form generator that takes a JSON configuration, built by consumer of library, and automatically builds out a form in Angular Material. I would like to allow user to pass custom attributes to each form field to suit there needs e.g. aria labels, flex-layout directives, etc... [ngAttr]="config.attrs" would be very useful. I am currently try to use elementRefs and viewChild but I am having some trouble getting it to behave as expected.

A feature like this seems more useful for code that builds code, but if the performance will take a big hit then I need to find another solution. My library is available here, and corresponding code is in the DynamicFormModule. I'll update if I find a good workaround.

Update
I got the custom attributes to show up in DOM by accessing a @ViewChild inside of an ngAfterViewInit and using Renderer2.setAttribute , but the flex-layout directives don't seem to have any effect. It seems like afterViewInit is happening after flex-layout does it's thing. Even if this did work, I am forced to use NativeElement when using this method, which warned against in documentation. Maybe the Ivy update will address these issues, maybe it won't.

I just wanted to make sure I voiced this as it is a good use case for the feature. Especially as users become better versed with angular and start building more generative code.

**Comment by GregKass** [comment]
+1 to @nayfin, for the basically the same reason. Also building JSON configurable form.
